src/main/java/org/apache/flink/graph/streaming/library/BipartitenessCheck.java

```diff
@@ -0,0 +1,124 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.graph.streaming.library;
+
+import org.apache.flink.api.common.functions.GroupReduceFunction;
+import org.apache.flink.api.common.functions.ReduceFunction;
+import org.apache.flink.graph.streaming.EdgesFold;
+import org.apache.flink.graph.streaming.WindowGraphAggregation;
+import org.apache.flink.graph.streaming.example.util.Candidates;
+import org.apache.flink.graph.streaming.example.util.SignedVertex;
+import org.apache.flink.types.NullValue;
+
+import java.io.Serializable;
+
+/**
+ * The bipartiteness check library to check whether an input graph is bipartite
+ * or not. A bipartite graph's vertices can be separated into two disjoint
+ * groups, such as no two nodes inside the same group is connected by an edge.
+ * The library uses the merge-tree abstraction of our graph streaming API.
+ *
+ * @param <K>  the vertex ID type
+ * @param <EV> the edge value type
+ */
+public class BipartitenessCheck<K extends Serializable, EV> extends WindowGraphAggregation<K, EV, Candidates, Candidates> implements Serializable {
+
+	private long mergeWindowTime;
+
+	public BipartitenessCheck(long mergeWindowTime) {
+		super(new updateFunction(), new combineFunction(), new Candidates(true), mergeWindowTime, false);
+	}
+
+	public static Candidates edgeToCandidate(long v1, long v2) throws Exception {
+		long src = Math.min(v1, v2);
+		long trg = Math.max(v1, v2);
+		Candidates cand = new Candidates(true);
+		cand.add(src, new SignedVertex(src, true));
+		cand.add(src, new SignedVertex(trg, false));
+		return cand;
+	}
+
+	@SuppressWarnings("serial")
+
+	/**
+	 * Implements EdgesFold Interface, applies a function to a vertex neighborhood
+	 * in the {@link GraphWindowStream#foldNeighbors(Object, EdgesFold)} method.
+	 *
+	 * @param <K> the vertex ID type
+	 */
+	public static class updateFunction<K extends Serializable> implements EdgesFold<Long, NullValue, Candidates> {
+
+		/**
+		 * Combines two edge values into one value of the same type.
+		 * The foldEdges function is consecutively applied to all edges of a neighborhood,
+		 * until only a single value remains.
+		 *
+		 * @param candidates the initial value and accumulator
+		 * @param v1         the vertex ID
+		 * @param v2         the neighbor's ID
+		 * @param edgeVal    the edge value
+		 * @return The data stream that is the result of applying the foldEdges function to the graph window.
+		 * @throws Exception
+		 */
```

Comment:
yes i used the javadoc of foldEdges method previously 
now changed it to :
Implements foldEdges method of EdgesFold interface for combining
two edges values into same type using merge method of the Candidates class.